### PR TITLE
Feature/pla 9753 expose cross team resource listing endpoints

### PR DESCRIPTION
**Pull request by crleblanc**
# Citrine Python PR

## Description 
As per PLA-9753 this PR gives users a way of listing published resources in a team, e.g. `team.dataset_ids.list_readable()`.  This returns a list of IDs.  This is not a great user experience, but our auth layer only records UIDs, so we would need additional APIs to hydrate these resources efficiently (e.g. getting a batch of resources).

There will be a PR for devkit to include an e2e for this functionality, once this PR is merged.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [X] I have added tests for 100% coverage
- [X] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [X] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/team.py [hunk 1]
```diff
@@ -33,6 +33,52 @@ def __str__(self):
...
+        return format_escaped_url(f'/teams/{str(self.team_id)}')
+
+    def _list_ids(self, action: str) -> List[str]:
+        query_params = {"domain": self._path(), "action": action}
+        return self.session.get_resource(f"/{self.resource_type}/authorized-ids",
+                                         params=query_params,
+                                         version=self._api_version)['ids']
+
+    def list_readable(self):
+        return self._list_ids(action="READ")
```
Comment (kroenlein) [inline src/citrine/resources/team.py:73]: This should be the module constant.
```suggestion
        return self._list_ids(action=READ)
```

File: src/citrine/resources/team.py [hunk 2]
```diff
@@ -33,6 +33,52 @@ def __str__(self):
...
+
+    def list_writeable(self):
+        return self._list_ids(action="WRITE")
```
Comment (kroenlein) [inline src/citrine/resources/team.py:76]: This should be the module constant.
```suggestion
        return self._list_ids(action=WRITE)
```

File: src/citrine/resources/team.py [hunk 3]
```diff
@@ -33,6 +33,52 @@ def __str__(self):
...
+
+    def list_shareable(self):
+        return self._list_ids(action="SHARE")
```
Comment (kroenlein) [inline src/citrine/resources/team.py:79]: This should be the module constant.
```suggestion
        return self._list_ids(action=SHARE)
```

File: src/citrine/resources/team.py [hunk 4]
```diff
@@ -33,6 +33,52 @@ def __str__(self):
...
+    def __init__(self,
+                 session: Session,
+                 team_id: Union[str, UUID],
+                 resource_type: str) -> None:
+        self.session = session
+        self.team_id = team_id
+        self.resource_type = resource_type
+
+    def _path(self) -> str:
+        return format_escaped_url(f'/teams/{str(self.team_id)}')
```
Comment (kroenlein) [inline src/citrine/resources/team.py:64]: I believe the f-string automatically casts as `str` by default
```suggestion
        return format_escaped_url(f'/teams/{self.team_id}')
```

**Review by kroenlein — CHANGES_REQUESTED** [review]
The build a collection then enumerate a set of methods doesn't quite align with my design aesthetic, but it's internally consistent and gets the job done.  I'll approve once you address the specific changes below.

( I'd probably have created one `resource_ids` method with the access and type arguments that take enumerations. )

**Review by kroenlein — APPROVED** [review]
Makes sense as written.  

Would make more sense to me grammatically if the syntax were:
```python
team.datasets.list_readable_ids()
```
but I appreciate that implementing that would be more complicated.

**Comment by crleblanc** [comment]
> Makes sense as written.
> 
> Would make more sense to me grammatically if the syntax were:
> 
> ```python
> team.datasets.list_readable_ids()
> ```
> 
> but I appreciate that implementing that would be more complicated.

I agree that this would be much cleaner, but this is not the DatasetCollection instance we get elsewhere, e.g. project.datasets.  I suspect this could get confusing for users who would expect a fully hydrated resource collection but instead get IDs back.

I wonder if we'll have to revisit this in the future and make batch-get endpoints to facilitate a proper resource collection?  The user workflow would be smoother.

**Pull request merged**